### PR TITLE
feat: menuitem: adds align icon prop and fixes list item rendering bug

### PR DESCRIPTION
**Pull request by dkilgore-eightfold**
## SUMMARY:
This is a part of ENG-45561

- Adds `alignIcon` prop in preparation for recursive dropdown menu support
- maps `IconSize` to `MenuSize`, omits size from `IconProps` via `MenuIconProps`
- Adds getIcon method to `MenuItemButton` and `MenuItemLink`
- Ensure `disabled` from props is read by `MenuItemLink`
- Adds `wrap` prop to `MenuItem` and RTL styles to enable menu item text wrapping
- Ensures `menuitem` `<li>` content is focusable and take the full width and height possible for greater click target
- Maps `MenuSize` and `IconSize` before the `iconProps` spread to ensure default size is the same as `Menu`
- Fixes `<li>` rendering bug
- Alphabetizes props
- Reduces the size of the `Dropdown` border radius by `4px` to better match systems design Figma
- Fixes `Dropdown` drift into position bug caused by unnecessary transition property on `.dropdown-wrapper`
- Add Jira identifier to TODO in `dropdown.module.scss`
- Adds `Menu` and `Dropdown` unit tests

Right aligned icon to enable sub menu visual cue:
alignIcon

Preserved and improved UI:
https://user-images.githubusercontent.com/99700808/225406107-bdedeca1-6096-4af6-84e4-3f6b60718bc1.mp4

## JIRA TASK (Eightfold Employees Only):
ENG-45561

## CHANGE TYPE:

- [X] Bugfix Pull Request
- [X] Feature Pull Request

## TEST COVERAGE:
- [ ] Tests for this change already exist
- [X] I have added unittests for this change

## TEST PLAN:
Pull the PR branch and run `yarn` and `yarn storybook`. Verify the `Dropdown`, `Menu` and `Select` stories behave as expected.

### CODE REVIEW

File: src/components/Menu/MenuItem/MenuItemButton/MenuItemButton.tsx [hunk 1]
```diff
@@ -46,25 +50,51 @@ export const MenuItemButton: FC<MenuItemButtonProps> = ({
     },
   ]);
 
+  const handleOnClick = (
+    event: React.MouseEvent<HTMLButtonElement | MouseEvent>
+  ): void => {
+    if (disabled) {
+      event.preventDefault();
```
Comment (szhang-eightfold) [inline src/components/Menu/MenuItem/MenuItemButton/MenuItemButton.tsx:57]: Just a little curious, what would happen if it is disabled and we dont prevent default here?

File: src/components/Menu/MenuItem/menuItem.module.scss [hunk 2]
```diff
@@ -116,10 +121,16 @@
     }
   }
 
-  &[disabled] {
-    pointer-events: none;
+  &.disabled {
+    cursor: not-allowed;
     opacity: $disabled-alpha-value;
+    pointer-events: none;
```
Comment (szhang-eightfold) [inline src/components/Menu/MenuItem/menuItem.module.scss:127]: if we have `pointer-events: none`, do we still need the preventDefault?

File: src/components/Menu/MenuItem/MenuItemButton/MenuItemButton.tsx [hunk 3]
```diff
@@ -1,26 +1,29 @@
...
 
 import styles from '../menuItem.module.scss';
 
 export const MenuItemButton: FC<MenuItemButtonProps> = ({
-  iconProps,
-  text,
-  subText,
-  variant = MenuVariant.neutral,
-  size = MenuSize.medium,
+  active,
```
Comment (szhang-eightfold) [inline src/components/Menu/MenuItem/MenuItemButton/MenuItemButton.tsx:11]: also curious what is active used for in general? Just highlighting?

File: src/components/Menu/MenuItem/menuItem.module.scss (hunk 2 above)
Comment (dkilgore-eightfold) [reply src/components/Menu/MenuItem/menuItem.module.scss:127]: Yes, easily changed in the web inspector.

File: src/components/Menu/MenuItem/MenuItemButton/MenuItemButton.tsx (hunk 3 above)
Comment (dkilgore-eightfold) [reply src/components/Menu/MenuItem/MenuItemButton/MenuItemButton.tsx:11]: styles

File: src/components/Dropdown/dropdown.module.scss [hunk 4]
```diff
@@ -23,7 +23,7 @@
   background: var(--dropdown-background-color);
   padding: $space-xs;
   box-shadow: $shadow-object-m;
-  border-radius: $border-radius-xl;
+  border-radius: $border-radius-l; // TODO: Add DropdownSize type and handle mapping via trigger size, then L 24px, M 20px, S 16px border-radius determined by size of the trigger.
```
Comment (szhang-eightfold) [inline src/components/Dropdown/dropdown.module.scss:26]: should we put down a small jira link here as well for better tracking?

**Review by szhang-eightfold — APPROVED** [review]

File: src/components/Menu/MenuItem/MenuItemButton/MenuItemButton.tsx [hunk 5]
```diff
@@ -46,25 +54,52 @@ export const MenuItemButton: FC<MenuItemButtonProps> = ({
...
+          className={styles.menuOuterButton}
+          disabled={disabled}
+          tabIndex={tabIndex}
           {...rest}
-          onClick={() => onClick?.(value)}
+          onClick={handleOnClick}
         >
-          {iconProps && <Icon {...iconProps} />}
+          {iconProps && alignIcon !== MenuItemIconAlign.Right && (
+            <Icon size={menuSizeToIconSizeMap.get(size)} {...iconProps} />
```
Comment (ychhabra-eightfold) [inline src/components/Menu/MenuItem/MenuItemButton/MenuItemButton.tsx:87]: nit: omit size from iconProps

File: src/components/Menu/MenuItem/MenuItemButton/MenuItemButton.tsx [hunk 6]
```diff
@@ -46,25 +54,52 @@ export const MenuItemButton: FC<MenuItemButtonProps> = ({
...
-          className={styles.menuInnerButton}
+          className={styles.menuOuterButton}
+          disabled={disabled}
+          tabIndex={tabIndex}
           {...rest}
-          onClick={() => onClick?.(value)}
+          onClick={handleOnClick}
         >
-          {iconProps && <Icon {...iconProps} />}
+          {iconProps && alignIcon !== MenuItemIconAlign.Right && (
```
Comment (ychhabra-eightfold) [inline src/components/Menu/MenuItem/MenuItemButton/MenuItemButton.tsx:86]: nit
```suggestion
          {iconProps && alignIcon === MenuItemIconAlign.Left && (
```

File: src/components/Menu/MenuItem/MenuItemButton/MenuItemButton.tsx [hunk 7]
```diff
@@ -75,21 +110,28 @@ export const MenuItemButton: FC<MenuItemButtonProps> = ({
...
-          {counter && <span>{counter}</span>}
+    <li role={role} tabIndex={-1} className={menuItemClasses}>
+      <button
+        className={styles.menuItemButton}
+        disabled={disabled}
+        tabIndex={tabIndex}
+        {...rest}
+        onClick={handleOnClick}
+      >
+        {iconProps && alignIcon !== MenuItemIconAlign.Right && (
```
Comment (ychhabra-eightfold) [inline src/components/Menu/MenuItem/MenuItemButton/MenuItemButton.tsx:121]: nit: same as above

File: src/components/Menu/MenuItem/MenuItemButton/MenuItemButton.tsx [hunk 8]
```diff
@@ -46,25 +54,52 @@ export const MenuItemButton: FC<MenuItemButtonProps> = ({
...
+            <Icon size={menuSizeToIconSizeMap.get(size)} {...iconProps} />
+          )}
           <span className={styles.menuItemWrapper}>
             <span className={styles.itemText}>
               <span className={styles.label}>{text}</span>
             </span>
           </span>
+          {iconProps && alignIcon === MenuItemIconAlign.Right && (
+            <Icon size={menuSizeToIconSizeMap.get(size)} {...iconProps} />
+          )}
```
Comment (ychhabra-eightfold) [inline src/components/Menu/MenuItem/MenuItemButton/MenuItemButton.tsx:96]: This snippet is repeating multiple times. Let create a method to getIcon

File: src/components/Menu/MenuItem/MenuItemLink/MenuItemLink.tsx [hunk 9]
```diff
@@ -1,27 +1,31 @@
...
+  text,
+  variant = MenuVariant.neutral,
+  wrap = false,
   ...rest
 }) => {
   const menuItemClasses: string = mergeClasses([
     styles.menuItem,
-    styles.menuLink,
     {
+      [styles.menuItemRtl]: direction === 'rtl',
```
Comment (ychhabra-eightfold) [inline src/components/Menu/MenuItem/MenuItemLink/MenuItemLink.tsx:27]: handle disabled

**Review by ychhabra-eightfold — CHANGES_REQUESTED** [review]

**Review by ychhabra-eightfold — APPROVED** [review]

**Pull request merged**